dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -349,11 +361,34 @@ protected void generateNodeConfig(String logDir) {
         return dirList;
     }
 
+    protected static String getLocatorsList(String userKey) {
+        Set<String> keys = SnappyBB.getBB().getSharedMap().getMap().keySet();
+        String locatorsList = null;
+        Set<String> locatorHostPortList = new LinkedHashSet<>();
+        for (String key : keys) {
+            if (key.startsWith(userKey)) {
+                Log.getLogWriter().info("Key Found..." + key);
+                String value = (String) SnappyBB.getBB().getSharedMap().get(key);
+                locatorHostPortList.add(value);
+            }
+        }
+        if (locatorHostPortList.size() == 0) {
+            return "";
+        }
+        StringBuilder builder = new StringBuilder();
+        for (String entry : locatorHostPortList) {
+            builder.append(entry).append(",");
+        }
+        locatorsList = builder.substring(0, builder.lastIndexOf(","));
+        return locatorsList;
```

Comment:
No need to double scan the string. simply use org.apache.commons.lang.StringUtils.join(Collection, seperator)